src/eventhandling-new.ts

```diff
@@ -0,0 +1,82 @@
+const log = require('./log');
+
+// wew use the mixin approach as described here:
+// https://mariusschulz.com/blog/mixin-classes-in-typescript
+type Constructor<T = {}> = new (...args: any[]) => T;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type EventHandler = (event?: any) => void;
+
+export function eventHandlingMixedIn<TBase extends Constructor>(Base: TBase, additionalEvents: string[] = []) {
```

Comment:
Is this a normal naming scheme for mixins in JS/TS? It sounds super weird to my ears.

Reply:
Do you mean the camel case for `eventHandlingMixedIn`? It's not a real class but a more or less normal function, so I named it accordingly, but I am not personally attached to it :) What would you prefer?

Reply:
I mean the "MixedIn". I'd prefer to keep it to just to what it was, and document it in JSDoc if necessary. As you already gave thumbs-up on refactoring it completely, it won't last much longer anyway. :)